filters/awssigner/awssigv4/sigv4_test.go

```diff
@@ -0,0 +1,359 @@
+		session           string
+		timeOfSigning     string
+		expectedSignature string
+		name              string
+	}{
+		{
+			accessKey:         "",
+			secret:            "some-invalid-secret",
+			session:           "some-invalid-session",
+			timeOfSigning:     "2012-11-01T22:08:41+00:00",
+			expectedSignature: "",
+			name:              "No_access_key_supplied",
+		},
+		{
+			accessKey:         "some-invalid-accesskey",
+			secret:            "",
+			session:           "some-invalid-session",
+			timeOfSigning:     "2012-11-01T22:08:41+00:00",
+			expectedSignature: "",
+			name:              "No_secret_key_provided",
+		},
+		{
+			accessKey:         "some-access-key",
+			secret:            "some-invalid-secret",
+			session:           "",
+			timeOfSigning:     "2012-11-01T22:08:41+00:00",
+			expectedSignature: "",
+			name:              "No_session_key_supplied",
+		},
+		{
+			accessKey:         "some-access-key",
+			secret:            "some-invalid-secret",
+			session:           "some-invalid-session",
+			timeOfSigning:     "",
+			expectedSignature: "",
+			name:              "No_time_of_signing_supplied",
+		},
+		{
+			accessKey:         "some-access-key",
+			secret:            "some-invalid-secret",
+			session:           "some-invalid-session",
+			timeOfSigning:     "2012-11-01T22:08:41+00",
+			expectedSignature: "",
+			name:              "incorrect_format_of_time_of_signing_supplied",
+		},
+		{
+			accessKey:         "AKID",
+			secret:            "SECRET",
+			session:           "SESSION",
+			timeOfSigning:     "1970-01-01T00:00:00Z",
+			expectedSignature: "AWS4-HMAC-SHA256 Credential=AKID/19700101/us-east-1/dynamodb/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-amz-meta-other-header;x-amz-meta-other-header_with_underscore;x-amz-security-token;x-amz-target, Signature=a518299330494908a70222cec6899f6f32f297f8595f6df1776d998936652ad9",
+			name:              "all_headers_supplied",
+		},
+	}
+
+	for _, test := range tests {
+		expectedSig := test.expectedSignature
+		headers := &http.Header{}
+		headers.Add("x-amz-accesskey", test.accessKey)
+		headers.Add("x-amz-secret", test.secret)
+		headers.Add("x-amz-session", test.session)
+		headers.Add("x-amz-time", test.timeOfSigning)
+
+		headers.Add("X-Amz-Target", "prefix.Operation")
+		headers.Add("Content-Type", "application/x-amz-json-1.0")
+		headers.Add("X-Amz-Meta-Other-Header", "some-value=!@#$%^&* (+)")
+		headers.Add("X-Amz-Meta-Other-Header_With_Underscore", "some-value=!@#$%^&* (+)")
+		headers.Add("X-Amz-Meta-Other-Header_With_Underscore", "some-value=!@#$%^&* (+)")
+		ctx := buildfilterContext(sigV4.service, sigV4.region, "POST", strings.NewReader("{}"), "//example.org/bucket/key-._~,!@#$%^&*()", *headers)
+
+		sigV4.Request(ctx)
+
+		signature := ctx.Request().Header.Get(internal.AuthorizationHeader)
+		b, _ := io.ReadAll(ctx.Request().Body)
+		assert.Equal(t, string(b), "{}") // test that body remains intact
```

Comment:
that's the "magic" body but I don't really get the reference why this is true to have `"{}"` instead of `http.NoBody` , which looks like https://go.dev/play/p/gbdvCM6LSEB shows that it would make more sense to have.

Reply:
Thanks,  the idea here is to test that body set [here](https://github.com/zalando/skipper/pull/3070/files#diff-bd0ce9a40faf89a6a7d55444655b8b8a4d55f8a0d40a8b91f37e12fc65ef84f2R297) is still intact after filter is done. 
This is needed as we are reading the body and resetting it back in the filter. 
I understand it is confusing to set it as `{}`, I can maybe change this to something else like `{"a":"b"}.` . will implement

Reply:
> that's the "magic" body but I don't really get the reference why this is true to have `"{}"` instead of `http.NoBody` , which looks like https://go.dev/play/p/gbdvCM6LSEB shows that it would make more sense to have.

Thanks,  the idea here is to test that body set [here](https://github.com/zalando/skipper/pull/3070/files#diff-bd0ce9a40faf89a6a7d55444655b8b8a4d55f8a0d40a8b91f37e12fc65ef84f2R297) is still intact after filter is done. 
This is needed as we are reading the body and resetting it back in the filter. 
I understand it is confusing to set it as `{}`, I can maybe change this to something else like `{"a":"b"}.` . will implement

Reply:
ok got it, thanks. `{}` is fine